src/Neo/SmartContract/Native/TrimmedBlock.cs

```diff
@@ -125,9 +125,14 @@ StackItem IInteroperable.ToStackItem(IReferenceCounter referenceCounter)
                 Header.PrimaryIndex,
                 Header.NextConsensus.ToArray(),
 
-                // Block properties
-                Hashes.Length
-            ]);
+            };
+            if (Header.Version == (uint)BlockVersion.V1)
+                block.Add(Header.PrevStateRoot.ToArray());
+
+            // Block properties
+            block.Add(Hashes.Length);
```

Comment:
I'm still think that if we append at the end, we don't affect SC that checks the `Hashes.Length` without check the version